image/image_linux.go

```diff
@@ -808,6 +825,9 @@ var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Opt
 	}
 
 	fetchAsserts := func(sn, sysSn, kSn *seedwriter.SeedSnap) ([]*asserts.Ref, error) {
+		if err := s.validateSnapArchs([]*seedwriter.SeedSnap{sn}); err != nil {
+			return nil, err
+		}
```

Comment:
I think this can be done before we download the snap,  before https://github.com/snapcore/snapd/blob/caff4098afab75d7713d88484c67eb7971f8fa48/image/image_linux.go#L460 ?

Reply:
Ah yeah definitely! Thanks for finding this!